directory/directory_transport.go

```diff
@@ -157,7 +157,11 @@ func (ref dirReference) NewImageDestination(ctx context.Context, sys *types.Syst
 	if sys != nil {
 		compress = sys.DirForceCompress
 	}
-	return newImageDestination(ref, compress)
+	decompress := false
+	if sys != nil {
+		decompress = sys.DirForceDecompress
+	}
+	return newImageDestination(ref, compress, decompress)
```

Comment:
This is fine, but at this point it would be more consistent with other transports and cleaner, to pass all of `sys` down to `newImageDestination(sys, ref)`;  `dirReference` doesn’t have to care about the compression options.